trace/trace.go

```diff
@@ -123,73 +107,83 @@ func WithSpan(parent context.Context, s *Span) context.Context {
 	return context.WithValue(parent, contextKey{}, s)
 }
 
-// StartSpanOptions contains options concerning how a span is started.
-type StartSpanOptions struct {
+// StartOptions contains options concerning how a span is started.
+type StartOptions struct {
 	// RecordEvents indicates whether to record data for this span, and include
 	// the span in a local span store.
 	// Events will also be recorded if the span will be exported.
 	RecordEvents bool
-	Sampler      // if non-nil, the Sampler to consult for this span.
+
+	Sampler // if non-nil, the Sampler to consult for this span.
```

Comment:
nit: by embedding, you're giving StartOptions all the methods of Sampler, which feels wrong. Even though it stutters, it is safer to write `Sampler Sampler`.